server/src/main/java/org/elasticsearch/common/util/concurrent/ThreadContext.java

```diff
@@ -134,6 +135,47 @@ public StoredContext stashContext() {
         };
     }
 
+    /**
+     * Removes the specified transient headers from the current context. The removed transient headers can be
+     * restored by closing the returned {@link StoredContext}. If such a transient header does not exist before
+     * this method is called, it will also not exist after restore (i.e. restoring asserts the existence
+     * and the values for the transient headers, as they existed before invoking this method).
+     */
+    public StoredContext stashTransientContext(Collection<String> transientHeadersToStash) {
```

Comment:
Naming is hard. I'd personally suggest this method to be just called `stashTransientHeaders`. The difference is to drop the Context bit, which is the confusing bit for me, because the existing `stashContext` behaves differently in a subtle but important way: 
* For the child actions, `stashContext` assigns it to a default context, while this method assigns it the "current" context minus the transients. This is more similar to the behaviour of `newStoredContext`, which assigns the current context unchanged.
* When restoring, `stashContext` restores the original context, while this method only restore selected trainstent headers. The important difference here is that this method does *not* restore anything else.

Reply:
I went with `stashTransientHeaders` , thanks for suggesting.